modules/integration/tests-integration/tests-amqp/src/test/java/org/wso2/mb/integration/tests/amqp/functional/TenantDeadLetterChannelTestCase.java

```diff
@@ -0,0 +1,214 @@
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.andes.stub.AndesAdminServiceBrokerManagerAdminException;
+import org.wso2.carbon.andes.stub.admin.types.Queue;
+import org.wso2.carbon.authenticator.stub.LoginAuthenticationExceptionException;
+import org.wso2.carbon.authenticator.stub.LogoutAuthenticationExceptionException;
+import org.wso2.carbon.automation.engine.FrameworkConstants;
+import org.wso2.carbon.automation.engine.context.AutomationContext;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.test.utils.axis2client.ConfigurationContextProvider;
+import org.wso2.carbon.integration.common.utils.LoginLogoutClient;
+import org.wso2.carbon.user.mgt.stub.UserAdminUserAdminException;
+import org.wso2.mb.integration.common.clients.AndesClient;
+import org.wso2.mb.integration.common.clients.configurations.AndesJMSConsumerClientConfiguration;
+import org.wso2.mb.integration.common.clients.configurations.AndesJMSPublisherClientConfiguration;
+import org.wso2.mb.integration.common.clients.exceptions.AndesClientConfigurationException;
+import org.wso2.mb.integration.common.clients.exceptions.AndesClientException;
+import org.wso2.mb.integration.common.clients.operations.clients.AndesAdminClient;
+import org.wso2.mb.integration.common.clients.operations.utils.AndesClientConstants;
+import org.wso2.mb.integration.common.clients.operations.utils.AndesClientUtils;
+import org.wso2.mb.integration.common.clients.operations.utils.ExchangeType;
+import org.wso2.mb.integration.common.clients.operations.utils.JMSAcknowledgeMode;
+import org.wso2.mb.integration.common.utils.backend.MBIntegrationBaseTest;
+import org.xml.sax.SAXException;
+
+import javax.jms.JMSException;
+import javax.naming.NamingException;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.xpath.XPathExpressionException;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.rmi.RemoteException;
+
+/**
+ * This test case contains test to check if messages goes to correct
+ * tenants dead letter channel.
+ */
+public class TenantDeadLetterChannelTestCase extends MBIntegrationBaseTest {
+
+    /**
+     * The default andes acknowledgement wait timeout.
+     */
+    private String defaultAndesAckWaitTimeOut = null;
+
+    /**
+     * Initializes the test case.
+     *
+     * @throws XPathExpressionException
+     * @throws java.rmi.RemoteException
+     * @throws org.wso2.carbon.user.mgt.stub.UserAdminUserAdminException
+     */
+    @BeforeClass(alwaysRun = true)
+    public void init() throws XPathExpressionException, RemoteException,
+                              UserAdminUserAdminException {
+        super.init(TestUserMode.SUPER_TENANT_USER);
+        AndesClientUtils.sleepForInterval(15000);
+
+        // Get current "AndesAckWaitTimeOut" system property.
+        defaultAndesAckWaitTimeOut = System.getProperty("AndesAckWaitTimeOut");
```

Comment:
Could we set this string to a pre-defined constant ? i've seen it being used everywhere. Easier to maintain if we can define all system properties used in test cases in a single place.